RuriLib/Blocks/Interop/Methods.cs

```diff
@@ -39,15 +40,57 @@ public static string ShellCommand(BotData data, string executable, string argume
          * These are not blocks, but they take BotData as an input. The ScriptBlockInstance will take care
          * of writing C# code that calls these methods where necessary once it's transpiled.
          */
-
-        public static async Task<T> InvokeNode<T>(BotData data, string scriptFile, object[] parameters)
+        public static async Task<T> InvokeNode<T>(BotData data, string scriptOrFile, object[] parameters, bool isScript = false, string scriptHash = null)
         {
             data.Logger.LogHeader();
-            var result = await StaticNodeJSService.InvokeFromFileAsync<T>(scriptFile, null, parameters, data.CancellationToken).ConfigureAwait(false);
+            T result;
+
+            if (isScript && !string.IsNullOrEmpty(scriptHash))
```

Comment:
If `isScript` is `true` but `scriptHash` is null or empty, it will try to (wrongly) invoke from file. This should never happen in our case, but it can happen in general depending on who calls the method, so we need to handle it. It can be rewritten as
```cs
if (isScript)
```
and `InvokeFromStringOrCache<T>` should expect a nullable string as the `scriptHash`, since the hash is not mandatory (even though we expect it to be there in our case for caching).